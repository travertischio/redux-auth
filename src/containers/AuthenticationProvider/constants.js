/*
 *
 * AuthenticationProvider constants
 *
 */

export const AUTH_KEY = 'auth';
export const LAST_ACTIVE_KEY = 'authLastActive';
export const ENCRYPT_SECRET_KEY = navigator.userAgent;

export const TOKEN_STATUS_VALID = 'valid';
export const TOKEN_STATUS_INVALID = 'invalid';
export const TOKEN_STATUS_AWAITING_SECOND_FACTOR = 'awaiting_second_factor';

export const SET_TOKEN_DATA_ACTION = 'app/AuthenticationProvider/SET_TOKEN_DATA';
export const CLEAR_TOKEN_DATA_ACTION = 'app/AuthenticationProvider/CLEAR_TOKEN_DATA';
export const MARK_TOKEN_AS_INVALID_ACTION = 'app/AuthenticationProvider/MARK_TOKEN_AS_INVALID';
export const EXTEND_TOKEN_LIFETIME_ACTION = 'app/AuthenticationProvider/EXTEND_TOKEN_LIFETIME';
export const MARK_AUTHENTICATION_PROVIDER_AS_READY_ACTION = 'app/AuthenticationProvider/MARK_TOKEN_AS_REFRESHED';
export const SET_USER_DATA_ACTION = 'app/AuthenticationProvider/SET_USER_DATA';
export const CLEAR_USER_DATA_ACTION = 'app/AuthenticationProvider/CLEAR_USER_DATA';
export const TWO_FACTOR_SEND_CODE_ACTION = 'app/AuthenticationProvider/TWO_FACTOR_SEND_CODE';
export const TWO_FACTOR_SEND_CODE_SUCCESS_ACTION = 'app/AuthenticationProvider/TWO_FACTOR_SEND_CODE_SUCCESS';
export const TWO_FACTOR_SEND_CODE_FAILED_ACTION = 'app/AuthenticationProvider/TWO_FACTOR_SEND_CODE_FAILED';
export const SIGN_OUT_ACTION = 'app/AuthenticationProvider/SIGN_OUT';
export const SIGN_OUT_SUCCESS_ACTION = 'app/AuthenticationProvider/SIGN_OUT_SUCCESS';
export const SIGN_OUT_FAILED_ACTION = 'app/AuthenticationProvider/SIGN_OUT_FAILED';
export const SET_LAST_USER_TOKEN = 'app/AuthenticationProvider/SET_LAST_USER_TOKEN';
export const REQUIRE_CAPTCHA_ACTION = 'app/AuthenticationProvider/REQUIRE_CAPTCHA';
export const BLOCKED_ACCOUNT_ACTION = 'app/AuthenticationProvider/BLOCKED_ACCOUNT';
export const FAILED_AUTHENTICATION_RESPONSE_ACTION = 'app/AuthenticationProvider/FAILED_AUTHENTICATION_RESPONSE';
export const SUCCESS_AUTHENTICATION_RESPONSE_ACTION = 'app/AuthenticationProvider/SUCCESS_AUTHENTICATION_RESPONSE';
