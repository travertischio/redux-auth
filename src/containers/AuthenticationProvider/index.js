/*
 *
 * AuthenticationProvider
 *
 */

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import compose from 'recompose/compose';
import withContext from 'recompose/withContext';
import { selectHasTokenRefreshed, selectIsAuthenticated } from './selectors';
import { refreshTokenAction } from './actions';

class AuthenticationProvider extends PureComponent {

  componentDidMount() {
    this.props.refreshToken();
  }

  renderLoading() {
    return (
      <div>
        Loading...
      </div>
    );
  }

  renderChildren() {
    return (
      <div>
        {React.Children.only(this.props.children)}
      </div>
    );
  }

  render() {
    if (this.props.hasTokenRefreshed) {
      return this.renderChildren();
    }

    return this.renderLoading();
  }
}

AuthenticationProvider.propTypes = {
  refreshToken: PropTypes.func.isRequired,
  children: PropTypes.element.isRequired,
  hasTokenRefreshed: PropTypes.bool.isRequired,
};

const AuthenticationProviderContext = withContext(
  {
    isAuthenticated: PropTypes.bool,
  },
  (props) => ({
    isAuthenticated: props.isAuthenticated,
  })
);

const composedComponent = compose(AuthenticationProviderContext)(AuthenticationProvider);

const mapStateToProps = createStructuredSelector({
  hasTokenRefreshed: selectHasTokenRefreshed,
  isAuthenticated: selectIsAuthenticated,
});

const mappedDispatch = {
  refreshToken: refreshTokenAction,
};

export default connect(mapStateToProps, mappedDispatch)(composedComponent);
