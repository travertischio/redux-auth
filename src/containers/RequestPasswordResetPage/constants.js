/*
 *
 * RequestPasswordResetPage constants
 *
 */

export const REQUEST_PASSWORD_RESET_ACTION = 'app/RequestPasswordResetPage/REQUEST_PASSWORD_RESET';
export const REQUEST_PASSWORD_RESET_SUCCEED_ACTION = 'app/RequestPasswordResetPage/REQUEST_PASSWORD_RESET_SUCCEED';
export const REQUEST_PASSWORD_RESET_FAILED_ACTION = 'app/RequestPasswordResetPage/REQUEST_PASSWORD_RESET_FAILED';
export const DESTROY_PAGE_ACTION = 'app/RequestPasswordResetPage/DESTROY';
