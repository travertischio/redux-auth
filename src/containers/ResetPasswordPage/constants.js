/*
 *
 * ResetPasswordPage constants
 *
 */

export const RESET_PASSWORD_ACTION = 'app/ResetPasswordPage/RESET_PASSWORD';
export const RESET_PASSWORD_SUCCEED_ACTION = 'app/ResetPasswordPage/RESET_PASSWORD_SUCCEED';
export const RESET_PASSWORD_FAILED_ACTION = 'app/ResetPasswordPage/RESET_PASSWORD_FAILED';
export const DESTROY_PAGE_ACTION = 'app/ResetPasswordPage/DESTROY';
