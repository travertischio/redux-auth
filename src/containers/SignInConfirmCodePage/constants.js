/*
 *
 * SignInConfirmCodePage constants
 *
 */

export const CONFIRM_CODE_ACTION = 'app/SignInConfirmCodePage/CONFIRM_CODE';
export const CONFIRM_CODE_SUCCESS_ACTION = 'app/SignInConfirmCodePage/CONFIRM_CODE_SUCCESS';
export const CONFIRM_CODE_FAILED_ACTION = 'app/SignInConfirmCodePage/CONFIRM_CODE_FAILED';
export const DESTROY_PAGE_ACTION = 'app/SignInConfirmCodePage/DESTROY';
