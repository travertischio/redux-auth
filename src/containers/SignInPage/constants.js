/*
 *
 * SignInPage constants
 *
 */

export const SIGN_IN_ACTION = 'app/SignInPage/SIGN_IN';
export const SIGN_IN_SUCCEED_ACTION = 'app/SignInPage/SUCCEED';
export const SIGN_IN_FAILED_ACTION = 'app/SignInPage/FAILED';
export const DESTROY_PAGE_ACTION = 'app/SignInPage/DESTROY';
