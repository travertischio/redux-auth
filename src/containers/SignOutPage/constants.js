/*
 *
 * SignOutPage constants
 *
 */

export const SIGN_OUT_ACTION = 'app/SignOutPage/SIGN_OUT';
