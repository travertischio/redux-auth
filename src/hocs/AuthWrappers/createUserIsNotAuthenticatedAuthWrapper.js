import { selectIsAuthenticated } from '~/containers/AuthenticationProvider/selectors';
import { redirectActionWithSupportParamInQueryString } from '~/containers/AuthenticationProvider/actions';
import connectedReduxRedirect from './connectedReduxRedirect';

function createUserIsNotAuthenticatedAuthWrapper(redirectPath) {
  return connectedReduxRedirect({
    authenticatedSelector: isNotAuthenticatedSelector,
    redirectAction: redirectActionWithSupportParamInQueryString,
    allowRedirectBack: false,
    wrapperDisplayName: 'UserIsNotAuthenticated',
    redirectPath,
  });
}

function isNotAuthenticatedSelector(state) {
  const isAuthenticated = selectIsAuthenticated(state);

  return !isAuthenticated;
}

export default createUserIsNotAuthenticatedAuthWrapper;
