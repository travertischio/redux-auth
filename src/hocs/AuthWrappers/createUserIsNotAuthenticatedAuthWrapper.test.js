import React from 'react';
import { createComponentWithRouter } from 'react-unit-testing-utils';
import { TOKEN_STATUS_VALID } from '~/containers/AuthenticationProvider/constants';
import createUserIsNotAuthenticatedAuthWrapper from './createUserIsNotAuthenticatedAuthWrapper';

const PageComponent = () => <div>Page only for not authenticated users</div>;
const failureRedirectPath = 'some/page';
const UserIsNotAuthenticatedAuthWrapper = createUserIsNotAuthenticatedAuthWrapper(failureRedirectPath);
const PageOnlyForNotAuthenticatedUser = UserIsNotAuthenticatedAuthWrapper(PageComponent);

describe('UserIsNotAuthenticated with custom failure redirect path', () => {
  it('should be defined', () => {
    expect(PageOnlyForNotAuthenticatedUser).toBeDefined();
  });

  it('should not redner PageComponent when user is authenticated', () => {
    const initialState = {
      auth: {
        tokenData: {
          status: TOKEN_STATUS_VALID,
        },
        userData: {
          id: 1,
        },
      },
    };

    const { wrapper } = createComponentWithRouter(PageOnlyForNotAuthenticatedUser, initialState);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should redner PageComponent when user is not authenticated', () => {
    const initialState = {
      auth: {},
    };

    const { wrapper } = createComponentWithRouter(PageOnlyForNotAuthenticatedUser, initialState);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
