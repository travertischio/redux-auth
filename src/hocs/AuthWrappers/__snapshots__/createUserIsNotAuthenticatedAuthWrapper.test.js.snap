// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserIsNotAuthenticated with custom failure redirect path should not redner PageComponent when user is authenticated 1`] = `<main />`;

exports[`UserIsNotAuthenticated with custom failure redirect path should redner PageComponent when user is not authenticated 1`] = `
<main>
  <div>
    Page only for not authenticated users
  </div>
</main>
`;
